### PR TITLE
doc: added steps to install Qdrant in local dev env

### PR DESCRIPTION
**Pull request by moaz-mokhtar**
### Documentation changes

After many trials to build and run Qdrant in my local machine and Gitpod, I preferred to add a guidance for how to prepare local development environment to build and run Qdrant.

I did test on below:
- My VM on KVM
```shell
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.2 LTS
Release:	22.04
Codename:	jammy
```
- Gitpod session
```shell
$  lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
```

### CODE REVIEW

**Comment by generall** [comment]
Hi @moaz-mokhtar, thanks for assembling this doc!

Could you please move it to https://github.com/qdrant/qdrant/blob/master/docs/DEVELOPMENT.md file maybe?

**Comment by moaz-mokhtar** [comment]
> 
Thanks @generall  for reply.
Ok I will, but I think I will need to move all `## Usage` to get complete guidance. What do you think?

**Comment by moaz-mokhtar** [comment]
Check now

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -2,6 +2,85 @@
...
+
+    unzip protoc-$PROTOC_VERSION-linux-x86_64.zip -d $HOME/.local
+
+    export PATH="$PATH:$HOME/.local/bin"
+
+    # remove source file if not needed
+    rm protoc-protoc-$PROTOC_VERSION-linux-x86_64.zip
+    
+    # check insalled `protoc` version
+    protoc --version
```
Comment (generall) [inline docs/DEVELOPMENT.md:75]: Could you please add a comment about what version do you expect to see here?

My local setup says 

```
protoc --version
libprotoc 3.12.4
```

which might be confusing, as we were installing `PROTOC_VERSION=22.2`

Comment (moaz-mokhtar) [reply docs/DEVELOPMENT.md:75]: I'm not sure which minimum version, but I found that version of `protoc` which installed by `apt` as below is not meet the requirements of Qdrant.
Check Issue https://github.com/qdrant/qdrant/issues/1385.

For latest `protoc` version by `apt` is: 3.12.4
```
$ sudo apt install protobuf-compiler
...
$ protoc --version
libprotoc 3.12.4
```

But when installing it from source with latest release you will get below:
```
$ protoc --version
libprotoc 22.2
```

Comment (moaz-mokhtar) [reply docs/DEVELOPMENT.md:75]: Also I didn't test it on `docker` yet, but I'm confused how it is working on `docker`!

Comment (generall) [reply docs/DEVELOPMENT.md:75]: It looks like debian and ubuntu have different versions pinned in APT:

https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=protobuf-compiler&searchon=names
https://packages.debian.org/sid/protobuf-compiler

Comment (generall) [reply docs/DEVELOPMENT.md:75]: But also I am confused why version semantic is so different - `22.2` vs `3.12.4`. Is it different forks or what is the reason?

Comment (moaz-mokhtar) [reply docs/DEVELOPMENT.md:75]: Regarding versioning, I did investigate this and did read many resources but I have some confusion. 
Generally it is common in the Protocol Buffers.

- Official documentation said:
> Versioning
> 
> We changed our versioning scheme to enable more-nimble updates to language-specific parts of Protocol Buffers. In the new scheme, each language has its own major version that can be incremented independently of other languages, as covered later in this topic with the Python release. The minor and patch versions, however, will remain coupled. This allows us to introduce breaking changes into some languages without requiring a bump of the major version in languages that do not experience a breaking change.
> 
> The first instance of this new versioning scheme is the new version of the Python API, 4.21.0, which follows the preceding version, 3.20.1. Other language APIs will be released as 3.21.0

Link: https://protobuf.dev/news/2022-05-06/#versioning

- This is one of the issues related to this topic: https://github.com/protocolbuffers/protobuf/issues/11440

- Below is answer of BingAI:

> Question 1: Why versioning in Protocol Buffers is not linear. For example v3.20.2 and v21.6?
> > Protocol Buffers is a serialization format for packets of typed, structured data that are up to a few megabytes in size. The format is suitable for both ephemeral network traffic and long-term data storage. Protocol Buffers can be extended with new information without invalidating existing data or requiring code to be updated⁵. 
> > Google designed protobuf to be pretty forgiving with versioning: unexpected data is either stored as "extensions" (making it round-trip safe), or silently dropped, depending on the implementation. New fields are generally added as "optional", meaning that old data can be loaded successfully⁴. 
> > The versioning scheme of Protocol Buffers has changed to enable more-nimble updates to language-specific parts of Protocol Buffers. In the new scheme, each language has its own major version that can be incremented independently of other languages⁶.

> > Source: Conversation with Bing, 3/21/2023(1) Overview | Protocol Buffers Documentation - Google Developers. https://protobuf.dev/overview/ Accessed 3/21/2023.
> > (2) How does protocol buffer handle versioning? - Stack Overflow. https://stackoverflow.com/questions/8519381/how-does-protocol-buffer-handle-versioning Accessed 3/21/2023.
> > (3) Changes made on May 6, 2022 | Protocol Buffers Documentation. https://protobuf.dev/news/2022-05-06/ Accessed 3/21/2023.
> > (4) How does protocol buffer handle versioning? - Stack Overflow. https://stackoverflow.com/questions/8519381/how-does-protocol-buffer-handle-versioning Accessed 3/21/2023.
> > (5) protocol buffers - gRPC / Protobuf interface versioning - Stack Overflow. https://stackoverflow.com/questions/40519610/grpc-protobuf-interface-versioning Accessed 3/21/2023.
> > (6) Protobuf: Explanation, Benefits + Tutorial - IONOS. https://www.ionos.com/digitalguide/websites/web-development/protocol-buffers-explained/ Accessed 3/21/2023.

> Question 2: What do you mean enable more-nimble updates to language-specific parts of Protocol Buffers?
> > The versioning scheme of Protocol Buffers has changed to enable more-nimble updates to language-specific parts of Protocol Buffers. In the new scheme, each language has its own major version that can be incremented independently of other languages. This means that changes to one language's implementation of Protocol Buffers will not affect other languages' implementations.

**Comment by moaz-mokhtar** [comment]
How to fix latest GithubAction check: "Integration tests / test-consensus (pull_request)" ?

File: docs/DEVELOPMENT.md [hunk 2]
```diff
@@ -2,6 +2,85 @@
...
+
+    # curl `proto` source file 
+    curl -LO https://github.com/protocolbuffers/protobuf/releases//download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip
+
+    unzip protoc-$PROTOC_VERSION-linux-x86_64.zip -d $HOME/.local
+
+    export PATH="$PATH:$HOME/.local/bin"
+
+    # remove source file if not needed
+    rm protoc-protoc-$PROTOC_VERSION-linux-x86_64.zip
```
Comment (moaz-mokhtar) [inline docs/DEVELOPMENT.md:72]: This wrong which should be `rm protoc-$PROTOC_VERSION-linux-x86_64.zip`

**Review by generall — APPROVED** [review]

**Pull request merged**